### PR TITLE
Solution class new method get_solution

### PR DESCRIPTION
**Pull request by SebGue**
The new method get_solution of Solution classes can be used to return a copy of the respective solution with a reduced set of indices e.g. the indices of a specifc group.
With this PR get_group of MeshSolution class now maintain the class of the solutions (instead of converting the solutions to SolutionMat class).

### CODE REVIEW

**Comment by SebGue** [comment]
@Superomeg4 Could you review the changes please?

File: pyleecan/Methods/Mesh/MeshSolution/get_group.py [hunk 1]
```diff
@@ -93,47 +94,16 @@ def get_group(self, group_names):
...
-            new_sol = SolutionMat(
-                label=label_sol,
-                type_cell=type_cell_sol,
-                field=new_field_sol,
-                indice=node_indice,
-                axis_name=axis_name,
-                axis_size=axis_size,
-            )
-            sol_list.append(new_sol)
+            new_indices = node_indice
```
Comment (Superomeg4) [inline pyleecan/Methods/Mesh/MeshSolution/get_group.py:100]: It was probably an error from previous modification, but it is the 1st occurence of node_indice.

File: pyleecan/Methods/Mesh/SolutionData/get_solution.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+from numpy import take
+from SciDataTool import Data1D
+
+
+def get_solution(self, indice=None):
+    """Return a copy of the solution with the option to only include specified indice.
+
+    Parameters
+    ----------
+    self : SolutionMat
```
Comment (Superomeg4) [inline pyleecan/Methods/Mesh/SolutionData/get_solution.py:11]: SolutionData

File: pyleecan/Methods/Mesh/SolutionData/get_solution.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                + "Respective indice will be skipped."
+            )
+
+        # skip indice that are not part of the solution
+        new_indice = [ii for ii in indice if ii in org_indice]
+
+        args = [name for name in axes_names]
+        args[ax_idx] += "=axis_data"
+
+        field_dict = solution.field.get_along(*args, axis_data={"indice": new_indice})
```
Comment (Superomeg4) [inline pyleecan/Methods/Mesh/SolutionData/get_solution.py:46]: A bit difficult to understand, maybe you can add a comment for the last 4 lines, but very smart way to proceed :-)

File: pyleecan/Methods/Mesh/SolutionVector/get_solution.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+from numpy import take
+from SciDataTool import Data1D
+
+
+def get_solution(self, indice=None):
+    """Return a copy of the solution with the option to only include specified indices.
+
+    Parameters
+    ----------
+    self : SolutionMat
```
Comment (Superomeg4) [inline pyleecan/Methods/Mesh/SolutionVector/get_solution.py:11]: SolutionVector

File: pyleecan/Methods/Mesh/SolutionVector/get_solution.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+                    + "Respective indice will be skipped."
+                )
+
+            # skip indice that are not part of the solution
+            new_indice = [ii for ii in indice if ii in org_indice]
+
+            args = [name for name in axes_names]
+            args[ax_idx] += "=axis_data"
+
+            field_dict = field.get_along(*args, axis_data={"indice": new_indice})
```
Comment (Superomeg4) [inline pyleecan/Methods/Mesh/SolutionVector/get_solution.py:48]: Same comment as SolutionData

File: Tests/Methods/Mesh/test_get_solution.py [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+        result = np_abs(expected - sol2.field).sum()
+        msg = "Wrong result: returned " + str(sol2) + ", expected: " + str(expected)
+        self.assertAlmostEqual(result, 0, msg=msg, delta=DELTA)
+
+        result = np_abs(expected - sol3.field).sum()
+        msg = "Wrong result: returned " + str(sol3) + ", expected: " + str(expected)
+        self.assertAlmostEqual(result, 0, msg=msg, delta=DELTA)
+
+        # result with explicit solution indieces
+        solution.indice = [0, 1, 2]
```
Comment (Superomeg4) [inline Tests/Methods/Mesh/test_get_solution.py:48]: If I understand correctly the purpose, maybe we could change the second part of the test to work with different indice triplet (such as a random 999, 2000, 11857)

File: pyleecan/Methods/Mesh/MeshSolution/get_group.py (hunk 1 above)
Comment (SebGue) [reply pyleecan/Methods/Mesh/MeshSolution/get_group.py:100]: node_indice is set in the 'for sep in sep_list:' loop starting in line 62 and in the 'if is_interface:' branch starting in line 85.

File: pyleecan/Methods/Mesh/SolutionData/get_solution.py (hunk 2 above)
Comment (SebGue) [reply pyleecan/Methods/Mesh/SolutionData/get_solution.py]: Fixed. Thank you!

File: pyleecan/Methods/Mesh/SolutionVector/get_solution.py (hunk 4 above)
Comment (SebGue) [reply pyleecan/Methods/Mesh/SolutionVector/get_solution.py]: Fixed. Thank you!

File: pyleecan/Methods/Mesh/MeshSolution/get_group.py (hunk 1 above)
Comment (Superomeg4) [reply pyleecan/Methods/Mesh/MeshSolution/get_group.py:100]: You are right, it was folded in the Github view. Sorry ;-)

File: Tests/Methods/Mesh/test_get_solution.py (hunk 6 above)
Comment (SebGue) [reply Tests/Methods/Mesh/test_get_solution.py]: I don't know if this is what you meant, but I set pseudo random indices. That also discovered a bug with my method. 
So, thanks!

Comment (Superomeg4) [reply Tests/Methods/Mesh/test_get_solution.py]: It was exactely what I meant. Great :-)

**Comment by Superomeg4** [comment]
The PR is OK for me.

**Pull request merged**